Sources/SwiftProtobuf/TextScanner.swift

```diff
@@ -15,55 +15,227 @@
 import Foundation
 import Swift
 
+private func fromHexDigit(_ c: UInt8) -> UInt8? {
+  if c >= 48 && c <= 57 {
```

Comment:
I may have glossed over this in the similar JSON PR, unfortunately :(  Swift has `UInt8(ascii:)` that takes a `UnicodeScalar` and produces the equivalent 8-bit int if it's a valid ASCII character. It's also able to optimize the whole thing down to just the single integer load instruction. I just tested it and `UInt8(ascii: "0")` becomes simple `mov al, 0x30` as expected.

Using that initializer instead of the raw ASCII would make this code a lot more readable/maintainable.